### PR TITLE
feat: new flight model matching new engine model with custom flaps and spoiler handling

### PR DESCRIPTION
**Pull request by aguther**
## Summary of Changes

This PR introduces a new flight model from @donstim that is adapted to the new engine model. It introduces distinct flap positions for 1 and 1+F. On top the spoilers have been given full drag for manual flight. They are now limited to half deployment when custom AP is engaged.

The spoiler handling is now fully custom made with the following features:
- extending spoilers disarms them
- extension logic for take-off and landing follow FCOM info
- partial extension according to FCOM is also supported

## Testing instructions

⚠️ The ballooning is known and will be improved, so please focus on the points mentioned below while testing.

Please test the following and provide feedback in contrast to current state on branch autopilot:
- are all displays showing correct info for spoilers and flaps?
- are warning generated correctly?
- do the checklists work?
- do the sounds work?
- flying and landing feeling in comparison
- Autoland on different runways in comparison (similar conditions)

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: A32NX/html_ui/Pages/VCockpit/Instruments/NavSystems/A320_Neo/A32NX_NavSystem.js [hunk 1]
```diff
@@ -2837,7 +2837,7 @@ class Warnings extends NavSystemElement {
     landingGearCallback() {
         const gear = !SimVar.GetSimVarValue("IS GEAR RETRACTABLE", "Boolean") || SimVar.GetSimVarValue("GEAR HANDLE POSITION", "Boolean");
         const throttle = SimVar.GetSimVarValue("L:A32NX_AUTOTHRUST_TLA:1", "number");
-        const flaps = SimVar.GetSimVarValue("L:A32NX_FLAPS_HANDLE_INDEX", "number");
+        contst flaps = SimVar.GetSimVarValue("L:A32NX_FLAPS_HANDLE_INDEX", "number");
```
Comment (tracernz) [inline A32NX/html_ui/Pages/VCockpit/Instruments/NavSystems/A320_Neo/A32NX_NavSystem.js:2840]: huh?

Comment (aguther) [reply A32NX/html_ui/Pages/VCockpit/Instruments/NavSystems/A320_Neo/A32NX_NavSystem.js]: forget it...I have overseen it. I do not understand why there was in any case a conflict on that line. Using the git function was big mess.

**Comment by MisterChocker** [comment]
The ecam symbols looks a bit misaligned
image

For reference IRL https://i.ytimg.com/vi/KMBzI0o6RIY/maxresdefault.jpg
or FSLABS https://i.gyazo.com/caa5e05cfe0fa6191d446429d1738a4b.png

**Comment by MisterChocker** [comment]
Also the wrong flaps notch is being marked blue. For Flaps 2 it indicates 3 and for 3 it does 4. Clean to 1 works fine.
And the indication for Flaps Full is missing entirely.

image
image

**Review by MisterChocker — CHANGES_REQUESTED** [review]
See comments above

**Pull request merged**